dangerfile.js

```diff
@@ -79,3 +79,59 @@ if (!isDependabotPR) {
     // Check Reviewer 2
     checkReviewerSection(reviewer2Section ? reviewer2Section[0] : null, 'Reviewer 2');
 }
+
+// README checks for Web Components (excluding core/testing projects)
+const readmeFiles = [...danger.git.created_files, ...danger.git.modified_files]
+    .filter((file) => /^packages\/components\/(?!pie-webc(?:-core|-testing)?\/)[^/]+\/README\.md$/.test(file));
+
+const checkReadmeStructure = async (filepath) => {
+    const diff = await danger.git.diffForFile(filepath);
+    const fileContent = diff.after;
+
+    const errors = [];
+
+    if (!/https:\/\/img\.shields\.io\/npm\/v\/@justeattakeaway\//.test(fileContent)) {
+        errors.push('Missing npm badge (https://img.shields.io/npm/v/@justeattakeaway/...).');
+    }
+
+    if (!/## Table of Contents/.test(fileContent)) {
+        errors.push('Missing "## Table of Contents" section.');
+    }
+
+    if (!/## Documentation/.test(fileContent)) {
+        errors.push('Missing "## Documentation" section.');
+    }
+
+    if (!/### Properties/.test(fileContent)) {
+        errors.push('Missing "### Properties" sub-section under Documentation.');
+    }
+    if (!/### Slots/.test(fileContent)) {
+        errors.push('Missing "### Slots" sub-section under Documentation.');
+    }
+    if (!/### Events/.test(fileContent)) {
+        errors.push('Missing "### Events" sub-section under Documentation.');
+    }
+    if (!/### CSS Variables/.test(fileContent)) {
+        errors.push('Missing "### CSS Variables" sub-section under Documentation.');
+    }
```

Comment:
i think some components don't have events and CSS variables section, would that throw a danger error if so?